core/domain/blog_domain.py

```diff
@@ -157,18 +157,18 @@ def validate(self, strict: bool = False) -> None:
                 raise utils.ValidationError(
                     'Expected Thumbnail filename should be a string,'
                     ' received %s' % self.thumbnail_filename)
+
             self.require_valid_thumbnail_filename(
                 self.thumbnail_filename, strict=strict)
 
-        if not isinstance(self.content, str):
-            raise utils.ValidationError(
-                'Expected contents to be a string, received: %s' % self.content)
-
-        if strict:
             self.require_valid_url_fragment(self.url_fragment)
             if not self.content:
                 raise utils.ValidationError('Content can not be empty')
 
+        if not isinstance(self.content, str):
+            raise utils.ValidationError(
+                'Expected contents to be a string, received: %s' % self.content)
+
```

Comment:
Why is this moved?

Reply:
There were 2 `if strict:`  blocks in the function. I just combined them to make it easier to understand the code.